hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -104,7 +109,8 @@ public String askQuestion(Question question)
 
             HttpRequest request = HttpRequest.newBuilder()
                     .uri(URI.create(config.getQuestionUrl()))
-                    .header("Content-Type", "application/json")
+                    .header(CONTENT_TYPE, "application/json")
+                    .header(USER_AGENT, applicationInfoProvider.getUserAgentData())
```

Comment:
I wonder if we could have solved this with a solution that is more fit to address cross-cutting concerns such as this one, e.g.: using an HttpRequestInterceptor?

I'm personally ok with this solution, but it is error-prone in the sense that all new APIs that are implemented rely on the developers to remember they have to manually add this user agent calculation into the request.

Reply:
I tried to use a more reliable solution but with `java.net.http.HttpClient` there's not much to do here.
I guess the best thing we could possibly do is some custom implementation (either for `java.net.http.HttpRequest` or maybe `java.net.http.HttpClient`) which would cover that. 
I'll push some more generic approach for this - let me know what you think.

Reply:
We could extend the `HttpClient#send` method to automatically decorate all requests, creating an e.g.: `HxHttpClient` class.

Similar to https://stackoverflow.com/questions/53215038/how-to-log-request-response-using-java-net-http-httpclient/55455589#55455589.